Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -0,0 +1,1765 @@
+                "please read integration documentation and follow the instructions"
+            )
+        client.get_folder_by_path(
+            client.folder_name, account, client.is_public_folder
+        ).test_access()
+    except ErrorFolderNotFound as e:
+        if "Top of Information Store" in str(e):
+            raise Exception(
+                "Success to authenticate, but user probably has no permissions to read from the specific folder."
+                "Check user permissions. You can try !ews-find-folders command to "
+                "get all the folders structure that the user has permissions to"
+            )
+
+    return "ok"
+
+
+def sub_main():
+    is_test_module = False
+    params = demisto.params()
+    client = EWSClient(**params)
+    args = prepare_args(demisto.args())
+    start_logging()
+    try:
+        command = demisto.command()
+        # commands that return a single note result
+        normal_commands = {
+            "ews-get-searchable-mailboxes": get_searchable_mailboxes,
+            "ews-move-item-between-mailboxes": move_item_between_mailboxes,
+            "ews-move-item": move_item,
+            "ews-delete-items": delete_items,
+            "ews-search-mailbox": search_items_in_mailbox,
+            "ews-get-contacts": get_contacts,
+            "ews-get-out-of-office": get_out_of_office_state,
+            "ews-recover-messages": recover_soft_delete_item,
+            "ews-create-folder": create_folder,
+            "ews-mark-item-as-junk": mark_item_as_junk,
+            "ews-find-folders": find_folders,
+            "ews-get-items-from-folder": get_items_from_folder,
+            "ews-get-items": get_items,
+            "ews-get-folder": get_folder,
+            "ews-expand-group": get_expanded_group,
+            "ews-mark-items-as-read": mark_item_as_read,
+        }
+
+        # commands that may return multiple results or non-note result
+        special_output_commands = {
+            "ews-get-attachment": fetch_attachments_for_message,
+            "ews-delete-attachment": delete_attachments_for_message,
+            "ews-get-items-as-eml": get_item_as_eml,
+        }
+        # system commands:
+        if command == "test-module":
+            is_test_module = True
+            demisto.results(test_module(client))
+        elif command == "fetch-incidents":
+            incidents = fetch_emails_as_incidents(client)
+            demisto.incidents(incidents)
+
+        # special outputs commands
+        elif command in special_output_commands:
+            demisto.results(special_output_commands[command](client, **args))  # type: ignore[operator]
+
+        # normal commands
+        else:
+            output = normal_commands[command](client, **args)  # type: ignore[operator]
+            return_outputs(*output)
+
+    except Exception as e:
+        time.sleep(2)
+        start_logging()
+        debug_log = log_stream.getvalue()  # type: ignore
```

Comment:
add error code here

Reply:
It's there

Reply:
where? its just `# type: ignore`

Reply:
Oh thought you meant the http error code, nm - I'll add it